tools/circt-rtl-sim/circt-rtl-sim.py.in

```diff
@@ -79,6 +79,60 @@ class Questa:
         return subprocess.run(cmd + simargs.split())
 
 
+class Vivado:
+    """Run and compile funcs for Vivado."""
+
+    DefaultDriver = "driver.sv"
+
+    def __init__(self, path, args):
+        self.path = path
+        self.args = args
+
+    def compile(self, sources):
+        sources = filter(
+            lambda fn: not (fn.endswith(".so") or fn.endswith(".dll")),
+            sources)
+        vlog = os.path.join(self.path, "xvlog")
+        args = [vlog, "-sv"] + list(sources)
+        rc = subprocess.run(args)
+        if rc.returncode != 0:
+            return rc
+
+        elab = os.path.join(self.path, "xelab")
+        if self.args.no_default_driver:
+            top = self.args.top
+        else:
+            top = "driver"
+        args = [elab, top, "-debug", "typical"]
+        return subprocess.run(args)
+
+    def run(self, cycles, simargs):
+        xsim = os.path.join(self.path, "xsim")
+
+        if self.args.no_default_driver:
+            top = self.args.top
+        else:
+            top = "driver"
+
+        args = [xsim, top]
+
+        script = "script.tcl"
+        with open(script, "w") as f:
+            commands = "run -all\n"
+            if not self.args.gui:
```

Comment:
I originally didn't use the TCL script option, and instead used the -R option, which is like "run all; quit". To support the --gui flag, I think the only way to _not_ quit is by using the TCL approach.